client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,461 @@
+		"device-key",
+	}
+)
+
+type ModelFormatter interface {
+	LongPublisher(storeAccountID string) string
+	GetEscapedDash() string
+}
+
+// modelAssertJSON is a helper to write out a single assertion in json
+// format that matches how it's done in api_model.go. We have stolen
+// the structure from there.
+type modelAssertJSON struct {
+	Headers map[string]interface{} `json:"headers,omitempty"`
+	Body    string                 `json:"body,omitempty"`
+}
+
+type PrintModelAssertionOptions struct {
+	TermWidth int
+	AbsTime   bool
+	Verbose   bool
+	Assertion bool
+}
+
+func fmtTime(t time.Time, abs bool) string {
+	if abs {
+		return t.Format(time.RFC3339)
+	}
+	return timeutil.Human(t)
+}
+
+func printVerboseAssertionHeaders(w *tabwriter.Writer, options PrintModelAssertionOptions, assertion asserts.Assertion) error {
+	allHeadersMap := assertion.Headers()
+	for _, headerName := range niceOrdering {
+		invalidTypeErr := fmt.Errorf("invalid type for %q header", headerName)
+
+		headerValue, ok := allHeadersMap[headerName]
+		// make sure the header is in the map
+		if !ok {
+			continue
+		}
+
+		// switch on which header it is to handle some special cases
+		switch headerName {
+		// list of scalars
+		case "required-snaps", "system-user-authority":
+			headerIfaceList, ok := headerValue.([]interface{})
+			if !ok {
+				// system-user-authority can also appear as string
+				headerString, ok := headerValue.(string)
+				if ok {
+					fmt.Fprintf(w, "%s:\t%s\n", headerName, headerString)
+					continue
+				}
+				return invalidTypeErr
+			}
+			if len(headerIfaceList) == 0 {
+				continue
+			}
+
+			fmt.Fprintf(w, "%s:\t\n", headerName)
+			for _, elem := range headerIfaceList {
+				headerStringElem, ok := elem.(string)
+				if !ok {
+					return invalidTypeErr
+				}
+				// note we don't wrap these, since for now this is
+				// specifically just required-snaps and so all of these
+				// will be snap names which are required to be short
+				fmt.Fprintf(w, "  - %s\n", headerStringElem)
+			}
+
+		// timestamp needs to be formatted with fmtTime from the timeMixin
+		case "timestamp":
+			timestamp, ok := headerValue.(string)
+			if !ok {
+				return invalidTypeErr
+			}
+
+			// parse the time string as RFC3339, which is what the format is
+			// always in for assertions
+			t, err := time.Parse(time.RFC3339, timestamp)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "timestamp:\t%s\n", fmtTime(t, options.AbsTime))
+
+		// long string key we don't want to rewrap but can safely handle
+		// on "reasonable" width terminals
+		case "device-key-sha3-384":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return invalidTypeErr
+			}
+
+			switch {
+			case options.TermWidth > 86:
+				fmt.Fprintf(w, "device-key-sha3-384: %s\n", headerString)
+			case options.TermWidth <= 86 && options.TermWidth > 66:
```

Comment:
The first condition is not needed:
```suggestion
			case options.TermWidth > 66:
```

Reply:
Fixed!